main.tf

```diff
@@ -2709,6 +2709,160 @@ module "vpa" {
   tags = var.tags
 }
 
+
+################################################################################
+# Velero
+################################################################################
+locals {
+  velero_name            = "velero"
+  velero_service_account = try(var.velero.service_account_name, "${local.velero_name}-sa")
+}
+
+# https://github.com/vmware-tanzu/velero-plugin-for-aws#option-1-set-permissions-with-an-iam-user
+data "aws_iam_policy_document" "velero" {
+  statement {
+    actions = [
+      "ec2:CreateSnapshot",
+      "ec2:CreateSnapshots",
+      "ec2:CreateTags",
+      "ec2:CreateVolume",
+      "ec2:DeleteSnapshot"
+    ]
+    resources = [
+      "arn:${local.partition}:ec2:${local.region}:${local.account_id}:instance/*",
+      "arn:${local.partition}:ec2:${local.region}::snapshot/*",
+      "arn:${local.partition}:ec2:${local.region}:${local.account_id}:volume/*"
+    ]
+  }
+
+  statement {
+    actions = [
+      "ec2:DescribeSnapshots",
+      "ec2:DescribeVolumes"
+    ]
+    resources = ["*"]
+  }
+
+  statement {
+    actions = [
+      "s3:AbortMultipartUpload",
+      "s3:DeleteObject",
+      "s3:GetObject",
+      "s3:ListMultipartUploadParts",
+      "s3:PutObject",
+    ]
+    resources = ["arn:${local.partition}:s3:::${var.velero.backup_s3_bucket}/*"]
```

Comment:
Let's just ask for the bucket ARN - it may not exist in the current account

Reply:
The addon is setup this way today, I don’t think we should change, and bucket ARN don’t have account ID.
Also it can generate confusion about the variable, informing the name of the bucket vs. bucket ARN or both. We could add a try:

```
resources = [try (“${var.velero.backup_s3_bucket_arn/*}”, "arn:${local.partition}:s3:::${var.velero.backup_s3_bucket}/*")]
```

But like I said I would go to create the S3 and let the customer decide to use ours or inform theirs.